_posts/2024-09-26-managing-connector-offsets.md

```diff
@@ -0,0 +1,277 @@
+---
+layout: post
+title:  "Managing Connector Offsets"
+date: 2024-08-26
+author: kate_stanley
+---
+
+When building data pipelines using Kafka Connect, or replicating data using MirrorMaker, offsets are used to keep track of the flow of data.
+Sink connectors use Kafka's standard consumer offset mechanism, while source connectors store offsets in a custom format within a Kafka topic.
+
+To manage connector offsets, rather than directly interacting with the underlying Kafka topics, you should make use of the following endpoints from the Connect REST API:
+
+* `GET /connectors/{connector}/offsets` to list offsets
+* `PATCH /connectors/{connector}/offsets` to alter offsets
+* `DELETE /connectors/{connector}/offsets` to reset offsets
+
+Some common use-cases for managing connector offsets are:
+* skipping a poison record
+* replaying records
+* specifying a starting offset for a new connector
+
+From Strimzi 0.44 onwards you can manage connector offsets directly in the `KafkaConnector` and `KafkaMirrorMaker2` custom resources.
+This blog post steps through how you can use this new functionality.
+The process is very similar for both Kafka Connect and MirrorMaker, so we'll demonstrate how to do it with Kafka Connect and then explain how the process applies to MirrorMaker.
+
+### Before we begin
+
+If you want to follow along the steps in this blog post you need a Kubernetes cluster containing the Strimzi operator, a Kafka cluster, a Connect cluster, and a running connector.
+First run through the Strimzi [quickstart guide](https://strimzi.io/quickstarts/) to deploy your Strimzi operator and Kafka cluster.
+
+Once you have a Kafka cluster you can deploy Connect and a connector using the following commands:
+1. `kubectl apply -f https://strimzi.io/examples/latest/connect/kafka-connect-build.yaml -n kafka`
+2. `kubectl wait kafkaconnect/my-connect-cluster --for=condition=Ready --timeout=300s -n kafka`
+3. `kubectl annotate kafkaconnect my-connect-cluster strimzi.io/use-connector-resources=true -n kafka`
+4. `kubectl apply -f https://strimzi.io/examples/latest/connect/source-connector.yaml -n kafka`
+5. `kubectl wait kafkaconnector/my-source-connector --for=condition=Ready --timeout=300s -n kafka`
```

Comment:
Wondering if all or some of the steps would need a one line comment to explain what/why you are doing that step.